services/carbon-intensity-provider/adapters/handler-http/handler.go

```diff
@@ -0,0 +1,55 @@
+package handler_http
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	"github.com/informatik-mannheim/cmg-ss2025/services/carbon-intensity-provider/ports"
+)
+
+type Handler struct {
+	service ports.Api
+	rtr     mux.Router
+}
+
+var _ http.Handler = (*Handler)(nil)
+
+func NewHandler(service ports.Api) *Handler {
+
+	h := Handler{service: service, rtr: *mux.NewRouter()}
+	h.rtr.HandleFunc("/carbon-intensity-provider/{id}", h.handleGet).Methods("GET")
+	h.rtr.HandleFunc("/carbon-intensity-provider", h.handleSet).Methods("PUT")
```

Comment:
As of now, the API only exposes two GET endpoints:
`GET /carbon-intensity/{zone}` and `GET /carbon-intensity/zones.`

While having a service method like AddOrUpdateZone is useful for local initialization or testing, implementing a `PUT` endpoint to update zones externally is currently unnecessary and out of scope.

I suggest creating 2 different handler methods for both endpoints:
```suggestion
	h.rtr.HandleFunc("/carbon-intensity-provider/{id}", h.handleGetZone).Methods("GET")
	h.rtr.HandleFunc("/carbon-intensity-provider", h.handleGetAllZones).Methods("PUT")
```

Reply:
added the suggestion